app/src/main/java/com/example/instagramclone/activities/createUser/EmailViewModel.kt

```diff
@@ -1,48 +1,69 @@
 package com.example.instagramclone.activities.createUser
 
 import android.util.Log
-import android.widget.Toast
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
-import com.example.instagramclone.activities.showToast
+import com.example.instagramclone.data.EmailRepository
 import com.google.firebase.auth.FirebaseAuth
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.schedulers.Schedulers
 
-class EmailViewModel : ViewModel() {
+class EmailViewModel(val emailRepository: EmailRepository) : ViewModel() {
 
-    val liveData = MutableLiveData<Any>()
+    val liveData = MutableLiveData<State>()
 
-    private fun FirebaseAuth.fetchSignInMethodsForEmail(email: String, onSuccess: (List<String>) -> Unit){
-        fetchSignInMethodsForEmail(email)
-            .addOnCompleteListener{
-            if (it.isSuccessful){
-                onSuccess (it.result?.signInMethods?: emptyList<String>())
+    fun onNext(email: String){
+        liveData.value = State.ShowLoading
 
-            }
-            else{
-                liveData.value = State.Error(it.exception!!.message!!)
+        CompositeDisposable().add(
```

Comment:
Separate disposables in your BaseViewModel like this https://github.com/zhanlederzach/DinnerAppAndroid/blob/master/app/src/main/java/kz/myfood/utils/base/BaseViewModel.kt